lib/src/async_cache.dart

```diff
@@ -89,12 +89,16 @@ class AsyncCache<T> {
   }
 
   /// Removes any cached value.
-  void invalidate() {
+  Null invalidate() {
     _cachedValueFuture = null;
+    // TODO: This does not await, but probably should.
     _cachedStreamSplitter?.close();
     _cachedStreamSplitter = null;
     _stale?.cancel();
     _stale = null;
+
+    // TODO: This does not return a future, but probably should.
```

Comment:
I think we can leave this method as it was (although we can either file an issue or add a TODO).

This can continue to be a `void` return and we can rewrite `await foo.invalidate();` to

```dart
foo.invalidate();
await null;
```

Reply:
Ah, missed your commend that you had explicitly avoided this. FWIW I prefer it since it localizes the "fix" to where the await is. It's also slightly less "breaking" to later change from `void` to `Future<void>` than it would be to change from `Null` to `Future<void>`.

Reply:
Part of the problem is that if this code *should* be awaited, because it *does* do asynchronous work, then I don't want to migrate all code awaiting this to no longer do so, only to then fix it and have to migrate all other code back..

Reply:
We've decided to keep allowing `await void` for now right? Maybe we should punt on this and file an issue to do the proper fix of making it return a `Future`?

Reply:
We might still get this in in time. But its not a priority